tqdm/_tqdm_web.py

```diff
@@ -0,0 +1,155 @@
+    """
+    def display_javascript(fp, s):
+        fp.write('<script type="text/javascript">')
+        fp.write(s)
+        fp.write('</script>')
+
+    def display_html(fp, s):
+        fp.write(s)
+
+    # Fallback to text bar if there's no total
+    if not total:
+        return StatusPrinter(file)
+
+    fp = file
+    if not getattr(fp, 'flush', False):  # pragma: no cover
+        fp.flush = lambda: None
+
+    desc = "<h3>%s</h3>" % desc if desc else ''
+    html_id = 'tqdm_' + str(uuid.uuid4())
+
+    # Prepare IPython progress bar
+    display_javascript(fp, '$("[data-key=\'%s\']").parent().parent().remove()' % key)
+    display_html(fp, '''
+            <style>
+                .progress {
+                    text-align:center;
+                }
+                .progress > .progress-bar {
+                    transition-property: none;
+                }
+                .progress > .text {
+                    position: absolute;
+                    right: 0;
+                    left: 0;
+                }
+            </style>
+            %s
+            <div class="progress" id="%s" data-key="%s">
+                <div class="progress-bar progress-bar-success completed-part" style="width: 0%%"></div>
+                <div class="progress-bar progress-bar-warning running-part" style="width: 100%%"></div>
+                <span class="text">
+                    <span class="main">Starting...</span>
+                    <span class="extra"></span>
+                </span>
+            </div>
+            ''' % (desc, html_id, key))
+
+    def print_status(s='', close=False):
+        # Clear previous output (really necessary?)
+        # clear_output(wait=1)
+
+        # Get current iteration value from format_meter string
+        n = None
+        if s:
+            npos = s.find(r'/|/')  # because we use bar_format=r'{n}|...'
+            # Check that n can be found in s (else n > total)
+            if npos >= 0:
+                n = int(s[:npos])  # get n from string
+                s = s[npos+3:]  # remove from string
+
+                # Update bar with current n value
+                if n is not None:
+                    display_javascript(fp, '$("#%s > .completed-part").css("width", "%f%%")' % (html_id, n/total*100))
+                    display_javascript(fp, '$("#%s > .running-part").css("width", "%f%%")' % (html_id, total/100))
+
+        # Print stats
+        display_javascript(fp, '$("#%s > .text > .main").text("%s")' % (html_id, s))
+
+        # Special signal to close the bar
+        if close:
+            display_javascript(fp, '$("#%s").parent().parent().hide()' % html_id)
+
+    return print_status
+
+
+class tqdm_web(tqdm):  # pragma: no cover
+    """
+    Experimental CSS/JSS web-friendly tqdm!
+    """
+    def __init__(self, *args, **kwargs):
+
+        # Setup default output
+        if not kwargs.get('file', None) or kwargs['file'] == sys.stderr:
```

Comment:
thanks. looks like this was copied from the current notebook implementation.
On 18 May 2016 7:11 a.m., "o11c" <notifications@github.com> wrote:

> In tqdm/_tqdm_web.py
> <https://github.com/tqdm/tqdm/pull/94#discussion_r63649948>:
>
> > +
> > +        # Special signal to close the bar
> > +        if close:
> > +            display_javascript(fp, '$("#%s").parent().parent().hide()' % html_id)
> > +
> > +    return print_status
> > +
> > +
> > +class tqdm_web(tqdm):  # pragma: no cover
> > +    """
> > +    Experimental CSS/JSS web-friendly tqdm!
> > +    """
> > +    def __init__(self, *args, **kwargs):
> > +
> > +        # Setup default output
> > +        if not kwargs.get('file', None) or kwargs['file'] == sys.stderr:
>
> kwargs.get('file', sys.stderr) is sys.stderr
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/tqdm/tqdm/pull/94/files/ca411dd998394c72cd977d80a699247cd7b0ea9f#r63649948>
>